sync-language-files.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+/**
+ * Plugin Name: Planet4 - Sync Language Files
+ * Description: Monitors changes to PO files during save operations in Loco Translate plugin & sync the language files.
+ * Plugin URI: http://github.com/greenpeace/planet4-plugin-sync-lang-files
+ * Version: 1.0
+ *
+ * Author: Greenpeace International
+ * Author URI: http://www.greenpeace.org/
+ * Text Domain: planet4-sync-lang-files
+ *
+ * License:     GPLv3
+ * Copyright (C) 2018 Greenpeace International
+ */
+
+// Exit if accessed directly.
+defined( 'ABSPATH' ) || die( 'Direct access is forbidden !' );
+
+/**
+ * Class LocoWebhook
+ */
+class LocoWebhook {
+	/**
+	 * Language file path.
+	 *
+	 * @var string
+	 */
+	private $path;
+
+	/**
+	 * Hash of language file path.
+	 *
+	 * @var string
+	 */
+	private $hash;
+
+	/**
+	 * Circle CI api token.
+	 *
+	 * @var $cci_auth_token string
+	 */
+	private $cci_auth_token = '';
+
+	const CCI_BASE_URL     = 'https://circleci.com/api/v1.1/project';
+	const CCI_AUTH_URL     = self::CCI_BASE_URL . '/github/greenpeace/planet4-handbook/build';
+	const CCI_CALL_TIMEOUT = 10;              // Seconds after which the api call will timeout if not responded.
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $path language file path.
+	 */
+	public function __construct( $path ) {
+		$this->path = $path;
+		if ( file_exists( $path ) ) {
+			$this->hash = md5_file( $path );
+		}
+	}
+
+	/**
+	 * Sync language file in GCS bucket.
+	 */
+	public function ping() {
+		if ( file_exists( $this->path ) ) {
+			$hash = md5_file( $this->path );
+			if ( $hash !== $this->hash ) {
+				$this->hash = $hash;
+
+				// Get Circel CI API token from DB.
+				$settings          = get_option( 'p4_sync_ci_token' );
+				$cci_private_token = $settings['p4ci_api_token'] ?? '';
+				// Consume Circle CI API.
+				$this->api_call( $cci_private_token );
+			}
+		}
+	}
+
+	/**
+	 * Hook on the loco translate save action.
+	 */
+	public static function init() {
+		if ( 'save' === sanitize_text_field( $_POST['route'] ) ) {
+			$path = WP_CONTENT_DIR . '/' . sanitize_text_field( $_POST['path'] );
```

Comment:
Actually, it was my mistake I used the word sanitize in the first comment on this one.
For better security and in order to prevent directory traversal attacks we need to validate the 'path' key (not sanitize).
So, I was thinking something more like [validate_file](https://codex.wordpress.org/Function_Reference/validate_file)

The 'route' key does not really need sanitization either since it is validated against 'save' string.